impeller/renderer/backend/gles/pipeline_library_gles.cc

```diff
@@ -257,6 +257,19 @@ PipelineFuture<ComputePipelineDescriptor> PipelineLibraryGLES::GetPipeline(
   return future;
 }
 
+// |PipelineLibrary|
+void PipelineLibraryGLES::RemovePipelinesWithEntryPoint(
+    std::shared_ptr<const ShaderFunction> function) {
+  for (auto it = pipelines_.begin(); it != pipelines_.end();) {
```

Comment:
I'm never quite sure about iterator invalidation rules due to mutation while enumeration and the safer C++20 std::erase_if is unfortunately unavailable to us. Can we put an erase_if shim in //impeller/base somewhere and do the dumb thing for now where do do mutate while enumerate (separate allocations are OK). This may be safe as written but it is hard to read and brittle in case we change the container type.

Reply:
Will also dry up the code a lot.

Reply:
Done. LMK if this is more or less what you had in mind.